### PR TITLE
feat(builtin.lsp): implement builtin handlers for lsp.(incoming|outgoing)_calls

### LINKED ISSUE #863: Add LSP outgoing and incoming calls
**Is your feature request related to a problem? Please describe.**
I am using telescope as my primary interface for list-like LSP results. One thing that is missing is support for the `outgoing_calls` and `incoming_calls` methods provided by the neovim LSP implementation.

**Describe the solution you'd like**
Having telescope lists similar to what is available for other LSP results would be great.

**Describe alternatives you've considered**

### PR DESCRIPTION
**Pull request by wilriker**
Fixes #863

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ wiki.
 
 - [sharkdp/fd](https://github.com/sharkdp/fd) (finder)
 - [nvim-treesitter/nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) (finder/preview)
-- [neovim LSP]( https://neovim.io/doc/user/lsp.html) (picker)
+- [neovim LSP](https://neovim.io/doc/user/lsp.html) (picker)
```
Comment (Conni2461) [inline README.md:65]: please no unrelated changes.

Please open a new PR for the proposed readme changes

**Comment by Conni2461** [comment]
FYI we currently are not looking for more pickers. Ref: https://github.com/nvim-telescope/telescope.nvim/issues/1228

File: README.md (hunk 1 above)
Comment (wilriker) [reply README.md]: I force-push removed these unrelated changes.

**Comment by Conni2461** [comment]
We might add this. I just rebased, cleaned up and tested it

**Comment by Conni2461** [comment]
Thanks :)

I am merging this in the current dev branch, see #1938 for more infos
